src/main/kotlin/com/openlattice/search/renderers/AlprAlertEmailRenderer.kt

```diff
@@ -0,0 +1,188 @@
+                null
+            } else values.iterator().next().toString()
+
+        }
+
+        private fun getImage(imageUrlPath: String?, fileType: String): ByteArrayOutputStream? {
+            if (imageUrlPath == null) {
+                return null
+            }
+
+            var stream: ByteArrayOutputStream? = null
+
+            try {
+                val imageUrl = URL(imageUrlPath)
+                val bufferedImage = ImageIO.read(imageUrl)
+                val baos = ByteArrayOutputStream()
+                ImageIO.write(bufferedImage, fileType, baos)
+                baos.flush()
+                stream = baos
+            } catch (e: IOException) {
+                logger.error("Unable to load image for url {}", imageUrlPath, e)
+            }
+
+            return stream
+
+        }
+
+        private fun getMapImage(coords: String?): ByteArrayOutputStream? {
+            if (coords == null) {
+                return null
+            }
+
+            val latAndLon = coords.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+            if (latAndLon.size < 2) {
+                return null
+            }
+
+            val lat = latAndLon[0]
+            val lon = latAndLon[1]
+
+            val url = StringBuilder("https://api.mapbox.com/v4/mapbox.streets/pin-l-car+000(")
+                    .append(lon)
+                    .append(",")
+                    .append(lat)
+                    .append(")/")
+                    .append(lon)
+                    .append(",")
+                    .append(lat)
+                    .append(",15/600x600.png?access_token=")
+                    .append(PersistentSearchMessengerHelpers.mapboxToken)
+                    .toString()
+
+            return getImage(url, PNG)
+        }
+
+        private fun extractVehicleInfo(vehicleRead: SetMultimap<FullQualifiedName, Any>): Map<String, Any> {
+
+            val values = Maps.newHashMap<String, Any>()
+
+            val readDateTimeList = vehicleRead.get(READ_TIMESTAMP_FQN)
+
+            values["readDateTime"] = if (readDateTimeList.isEmpty()) "" else OffsetDateTime.parse(readDateTimeList.first().toString()).format(dateTimeFormatter)
+            values["hitType"] = getFirstValue(vehicleRead, HIT_TYPE_FQN)
+            values["cameraId"] = getFirstValue(vehicleRead, CAMERA_ID_FQN)
+            values["agencyId"] = getFirstValue(vehicleRead, AGENCY_NAME_FQN)
+            values["make"] = getFirstValue(vehicleRead, MAKE_FQN)
+            values["model"] = getFirstValue(vehicleRead, MODEL_FQN)
+            values["year"] = getFirstValue(vehicleRead, YEAR_FQN)
+            values["color"] = getFirstValue(vehicleRead, COLOR_FQN)
+            values["coordinates"] = getFirstValue(vehicleRead, LOCATION_COORDS_FQN)
+
+
+            return values.filterValues { it != null }
+        }
+
+        private fun extractVehicleImages(vehicleRead: SetMultimap<FullQualifiedName, Any>): List<ByteArrayAttachment> {
```

Comment:
This function will work and probably can't be changed until content-type is set on upload by piper.